sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -3187,6 +3187,13 @@ def __init__(self, expr, kwargs,
     self._grouping_indexes = grouping_indexes
     self._kwargs = kwargs
 
+    if (self._kwargs.get('dropna', True) is False and
+        self._ungrouped.proxy().index.nlevels > 1):
+      raise NotImplementedError(
+          "dropna=False does not work as intended in the Beam DataFrame API "
+          "when grouping on multiple columns or indexes (See BEAM-12495).")
```

Comment:
Reference the Pandas bug as well for increased visibility?

Reply:
I think that might be confusing since the upstream bug is _just_ when grouping by index levels, but our users will encounter it  even when grouping on columns since we move columns to the index.

Reply:
Note the pandas bug is referenced in the jira, so a motivated user will still come across it